.ado/jobs/universal.yml

```diff
@@ -1,147 +1,227 @@
+                    pathtoCustomTestAdapters: $(GoogleTestAdapterPath)
+                    searchFolder: $(Build.SourcesDirectory)/vnext/target/${{ matrix.BuildPlatform }}/${{ matrix.BuildConfiguration }}
+                    runTestsInIsolation: true
+                    platform: ${{ matrix.BuildPlatform }}
+                    configuration: ${{ matrix.BuildConfiguration }}
+                    publishRunAttachments: true
+                    collectDumpOn: onAbortOnly
+                    vsTestVersion: latest
+                  condition: and(succeeded(), not(startsWith(variables.BuildPlatform, 'arm')))
+
+                - task: VSTest@2
+                  displayName: Run Universal Unit Tests (UWP)
+                  timeoutInMinutes: 5 # Set smaller timeout , due to hangs
+                  inputs:
+                    testSelector: testAssemblies
+                    testAssemblyVer2: |
+                      Microsoft.ReactNative.Managed.UnitTests/Microsoft.ReactNative.Managed.UnitTests/Microsoft.ReactNative.Managed.UnitTests.build.appxrecipe
+                    searchFolder: $(Build.SourcesDirectory)/vnext/target/${{ matrix.BuildPlatform }}/${{ matrix.BuildConfiguration }}
+                    runTestsInIsolation: true
+                    platform: ${{ matrix.BuildPlatform }}
+                    configuration: ${{ matrix.BuildConfiguration }}
+                    publishRunAttachments: true
+                    codeCoverageEnabled: true
+                    collectDumpOn: onAbortOnly
+                    vsTestVersion: latest
+                  condition: and(succeeded(), not(startsWith(variables.BuildPlatform, 'arm')))
+
+                - task: VSTest@2
+                  displayName: Run Universal Unit Tests (NetCore)
+                  timeoutInMinutes: 5 # Set smaller timeout , due to hangs
+                  inputs:
+                    testSelector: testAssemblies
+                    testAssemblyVer2: |
+                      Microsoft.ReactNative.Managed.CodeGen.UnitTests/Microsoft.ReactNative.Managed.CodeGen.UnitTests.dll
+                    searchFolder: $(Build.SourcesDirectory)/vnext/target/${{ matrix.BuildPlatform }}/${{ matrix.BuildConfiguration }}
+                    runTestsInIsolation: true
+                    platform: ${{ matrix.BuildPlatform }}
+                    configuration: ${{ matrix.BuildConfiguration }}
+                    publishRunAttachments: true
+                    codeCoverageEnabled: true
+                    collectDumpOn: onAbortOnly
+                    vsTestVersion: latest
+                  condition: and(succeeded(), eq(variables.BuildPlatform, 'x64'))
+
+    # This job is the one that accumulates the spread out build tasks into one dependency
+    - job: UniversalBuild
+      condition: ne( dependencies.Setup.outputs['checkPayload.shouldSkipPRBuild'], 'True' )
+      displayName: Universal Build 🏗
+      variables:
+        - template: ../variables/vs2019.yml
+      dependsOn: 
```

Comment:
I didn't do this for the desktop NuGet job, but I keep on noticing we try to run and fail that job when its dependents fail. Wondering if we can condition on the success of dependencies.

Reply:
Actually for this one since its aggregating, I don't think success is propagated. Wondering if we could do that.

Reply:
I have a seperate experiment using stages, so that we don't need to have custom conditions with manually having to deal with dependencies and let the default conditions work.

Reply:
#6623 This cleans up the conditions and the deps on Setup. It is not ideal yet since the nuget package is needed for the CLI test which ideally would be it's own stage..
I'm open to suggestions of groupings for that one.